### PR TITLE
ANSIENG-1990 Improve error handling deploying connectors

### PR DESCRIPTION
**Pull request by wadhwa1**
# Description

- Improve error handling while deploying connectors (Based on previous PR - https://github.com/confluentinc/cp-ansible/pull/490 + added error handling in create_new_connector)
- Fix kafka_connectors module to support ssl_enabled=true
- Updated molecule scenarios to test above - error handling of module + deploying connectors with ssl_enabled=true

Fixes # (issue)
GH #401  

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
https://jenkins.confluent.io/job/cp-ansible-on-demand/996/

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Review by utkarsh5474 — COMMENTED** [review]
Can we add a test to check the working of kafka connectors module having ssl_enabled?

File: molecule/archive-plain-rhel/verify.yml [hunk 1]
```diff
@@ -67,6 +67,113 @@
...
+        active_connectors_input: "{{ update_existing_connector }}"
+        expected_message: "Connectors added or updated: test-connector-1: connector configuration updated."
+
+    - import_role:
+        name: confluent.test
+        tasks_from: check_connector.yml
+      vars:
+        connect_url_input: "{{ kafka_connect_url }}"
+        active_connectors_input: "{{ add_bad_request_connector }}"
+        expected_message: "Connectors added or updated: test-connector-1: no configuration change, test-connector-2: ERROR error while updating configuration (HTTP Error 400: Bad Request)."
```
Comment (utkarsh5474) [inline molecule/archive-plain-rhel/verify.yml:158]: Here, are we trying to add test-connector2 for the first time or updating configurations of an already added test-connector2? 
Might be wrong but I believe it's the former, but the error message suggests it's the latter.

File: molecule/multi-ksql-connect-rhel/molecule.yml [hunk 2]
```diff
@@ -176,6 +176,8 @@ provisioner:
               tasks.max: "1"
               file: "/etc/kafka/connect-distributed.properties"
               topics: "test_topic"
+              key.converter: "org.apache.kafka.connect.json.JsonConverter"
```
Comment (utkarsh5474) [inline molecule/multi-ksql-connect-rhel/molecule.yml:179]: Personal doubt:
What do these properties help with? Any utility for us?

File: molecule/plain-rhel/verify.yml [hunk 3]
```diff
@@ -183,6 +183,120 @@
...
+
+    - import_role:
+        name: confluent.test
+        tasks_from: check_connector.yml
+      vars:
+        connect_url_input: "{{ kafka_connect_url }}"
+        active_connectors_input: "{{ delete_connector }}"
+        expected_message: "Connectors removed: test-connector-1. Connectors added or updated: test-connector-2: new connector added."
+
+    - import_role:
```
Comment (utkarsh5474) [inline molecule/plain-rhel/verify.yml:292]: Hey, this looks unrelated. Is this intended?

File: plugins/modules/kafka_connectors.py [hunk 4]
```diff
@@ -53,16 +53,21 @@
...
 
+RUNNING_STATE = "RUNNING"
+WAIT_TIME_BEFORE_GET_STATUS = 1  # seconds
+TIMEOUT_WAITING_FOR_TASK_STATUS = 30  # seconds
+
 
 def get_current_connectors(connect_url):
     try:
-        res = open_url(connect_url)
+        res = open_url(connect_url, validate_certs=False)
```
Comment (utkarsh5474) [inline plugins/modules/kafka_connectors.py:70]: Another self doubt - are we having validate_certs set to False across the script to handle the case of ssl enabled?

**Comment by wadhwa1** [comment]
> Can we add a test to check the working of kafka connectors module having ssl_enabled? If it's already there, that's awesome. Thanks for the PR. Some minor comments below.

Thanks @utkarsh5474 for reviewing the PR! :)
yup, **archive-plain-rhel** scenario has ssl_enabled=true, have added verification steps in the same to check kafka connectors module.

File: plugins/modules/kafka_connectors.py (hunk 4 above)
Comment (wadhwa1) [reply plugins/modules/kafka_connectors.py:70]: yes @utkarsh5474 , this is primarily to skip ssl certs verification in case of ssl enabled.
we'll re-visit this logic to pass certs and all when we deploy connectors with MTLS enabled.

File: molecule/plain-rhel/verify.yml (hunk 3 above)
Comment (wadhwa1) [reply molecule/plain-rhel/verify.yml:292]: this looks to be duplicate, will fix it, thanks!

File: molecule/multi-ksql-connect-rhel/molecule.yml (hunk 2 above)
Comment (wadhwa1) [reply molecule/multi-ksql-connect-rhel/molecule.yml:179]: yup, as there's no schema registry configured in this test, we need to explicitly mention key and value converter classes

File: molecule/archive-plain-rhel/verify.yml (hunk 1 above)
Comment (wadhwa1) [reply molecule/archive-plain-rhel/verify.yml]: checked again, fixed, thanks for pointing!

**Review by nsharma-git — APPROVED** [review]
